File: civitasJAVA/src/civitas/CasillaSorpresa.java
package civitas;

import java.util.ArrayList;

/**
 * @author Jesús Navarro Merino
 * @author José Alberto Gómez García
 */

public class CasillaSorpresa extends Casilla{

    private MazoSorpresas mazo;
    
    public CasillaSorpresa(MazoSorpresas _mazo,String nombre_introducido) {
        super(nombre_introducido);
        mazo=_mazo;
    }
    
    @Override
    void recibeJugador(int actual, ArrayList<Jugador> todos){
        if(jugadorCorrecto(actual,todos)){
            Sorpresa sorpresa=mazo.siguiente();
            informe(actual,todos);
            sorpresa.aplicarAJugador(actual, todos);
        }
    }
    
    @Override
    public String toString(){
        return super.toString();
    }
    
} //Fin de la clase CasillaSorpresa

File: civitasRUBY/lib/casilla_sorpresa.rb
# encoding:utf-8

module Civitas
  class CasillaSorpresa < Casilla
    
    attr_reader :nombre, :mazo
    
    def initialize(_nombre,_mazo)
      super(_nombre)
      @mazo=_mazo      
    end
    
    def recibeJugador(actual,todos)
      if(jugadorCorrecto(actual,todos))
        sorpresa=@mazo.siguiente
        informe(actual,todos)
        sorpresa.aplicarAJugador(actual,todos)
     end
    end
    
    def self.constructorCasillaSorpresa(_nombre,_mazo)
      new(_nombre,_mazo)
    end
    
    def toString
      super
    end
    
  end
end
